autoprofile.py

```diff
@@ -33,32 +35,106 @@ def __init__(self):
         self.name = _("Automatic Profile")
         self.bio_enabled = False
         self.name_enabled = False
+        self.pfp_enabled = False
+        self.pfp_remove_latest = None
         self.raw_bio = None
         self.raw_name = None
+        self.pfp_degree = 0
 
     async def client_ready(self, client, db):
         self.client = client
 
+    async def autopfpcmd(self, message):
+        """Rotates your profile picture every n seconds with x degrees, usage:
+           .autopfp <timeout> <degrees> <remove previous (last pfp)>
+
+           Timeout - seconds
+           Degrees - 60, -10, etc
+           Remove last pfp - true or false, boolean, case insensitive"""
+
+        msg = utils.get_args(message)
+        if len(msg) != 3:
+            return await utils.answer(message, _("<b>Autopfp requires three args. See the help for syntax.</b>"))
+
+        try:
+            timeout_autopfp = int(msg[0])
+        except ValueError as e:
+            logger.warning(str(e))
+            return await utils.answer(message, _("<b>Wrong time.</b>"))
+
+        try:
+            degrees = int(msg[1])
+        except ValueError as e:
+            logger.warning(str(e))
+            return await utils.answer(message, _("<b>Wrong degrees value.</b>"))
+
+        if msg[2].lower() == 'true':
+            delete_previous = True
+        elif msg[2].lower() == 'false':
+            delete_previous = False
+        else:
+            return await utils.answer(message, _("<b>Please pass true or false for previous pfp removing.</b>"))
+
+        me = await self.client.get_me()
+        pfp = await self.client.download_profile_photo(me, file=bytes)
+        raw_pfp = Image.open(BytesIO(pfp))
+
+        self.pfp_remove_latest = delete_previous
+        self.pfp_enabled = True
+        await utils.answer(message, "<b>Successfully enabled autopfp.</b>")
+
+        while self.pfp_enabled:
+            self.pfp_degree += degrees
+            rotated = raw_pfp.rotate(self.pfp_degree)
+            buf = BytesIO()
+            rotated.save(buf, format='JPEG')
+            bytes_img = buf.getvalue()
+
+            if self.pfp_remove_latest:
+                await self.client(functions.photos.DeletePhotosRequest(
+                    await self.client.get_profile_photos(await self.client.get_me(), limit=1)
+                ))
+
+            await self.client(functions.photos.UploadProfilePhotoRequest(
+                await self.client.upload_file(bytes_img)
```

Comment:
Substitute bytes_img for buf; its the same but then you can remove bytes_img and reduce the RAM usage